### PR TITLE
Run AT's against PKS on GCP instance

### LINKED ISSUE #202: Run AT's against PKS on GCP instance
Based on work done in https://github.com/spring-cloud/spring-cloud-dataflow-server-kubernetes/issues/347

### PR DESCRIPTION
**Pull request by chrisjs**
Resolves #202

### CODE REVIEW

**Comment by chrisjs** [comment]
some notes:
--
* the only thing that should differ between k8s "platforms" is the auth method, so the `kubernetes` `platform` is separated out into `gke` and `pks`, symlinked to `kubernetes-common` - all for the most part with the exception of `init/setenv.sh` 

* first passing test run pointing to the branch this PR is issued from: https://build.spring.io/browse/SCDT-PKS111

* env info is in the environment spreadsheet

after merge tasks:
--
- [ ] use official AT repo in CI plan
- [ ] add back notifications to CI plan
- [ ] update non-PKS plans to use "gke" as platform name rather than "kubernetes"
- [ ] re-enable plan

feel free just to drop a LGTM and i can take care of the merge and things in `after merge tasks`

File: README.adoc [hunk 1]
```diff
@@ -160,6 +160,24 @@ If you use a service account make sure to set the GOOGLE_APPLICATION_CREDENTIALS
...
+== Kubernetes (PKS)
+
+=== Pre-requisites
+
+* The `kubectl` command line tool needs to be installed. Installation information can be found at: https://kubernetes.io/docs/tasks/tools/install-kubectl/
+* The PKS command line tool needs to be installed and can be obtained from: https://network.pivotal.io/products/pivotal-container-service
+
+=== Configuration
+
+For Kubernetes, make sure you have the following environment variables exported.
```
Comment (sabbyanandan) [inline README.adoc]: Probably drop "For Kubernetes"

File: README.adoc [hunk 2]
```diff
@@ -160,6 +160,24 @@ If you use a service account make sure to set the GOOGLE_APPLICATION_CREDENTIALS
...
+
+* PKS_CLUSTER_NAME
+* PKS_ENDPOINT
+* PKS_USERNAME
+* PKS_PASSWORD
+
+NOTE: You can also set a KUBERNETES_NAMESPACE environment variable that specifies an existing namespace to use for the testing. If this is not specified the 'default' namespace will be used.
```
Comment (sabbyanandan) [inline README.adoc]: please add a comma after "specified"

File: pks/init/setenv.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+function pks_authenticate_and_target() {
+  [ -z "$PKS_CLUSTER_NAME" ] && { echo "Environment variable PKS_CLUSTER_NAME must be set"; exit 1; }
+  [ -z "$PKS_ENDPOINT" ] && { echo "Environment variable PKS_ENDPOINT must be set"; exit 1; }
+  [ -z "$PKS_USERNAME" ] && { echo "Environment variable PKS_USERNAME must be set"; exit 1; }
+  [ -z "$PKS_PASSWORD" ] && { echo "Environment variable PKS_PASSWORD must be set"; exit 1; }
+
+  echo "Using PKS cluster: $PKS_CLUSTER_NAME"
+
+  pks login -a $PKS_ENDPOINT -u $PKS_USERNAME -p $PKS_PASSWORD --skip-ssl-validation
```
Comment (sabbyanandan) [inline pks/init/setenv.sh]: What if someone runs the script against a cluster with valid certs? Do we need to add a note about `--skip-ssl-validation`?

File: pks/init/setenv.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  pks get-credentials $PKS_CLUSTER_NAME
+  kubectl config use-context $PKS_CLUSTER_NAME
+}
+
+# ======================================= FUNCTIONS END =======================================
+
+if ! command_exists pks; then
+  echo "You don't have the 'PKS' installed, please visit https://network.pivotal.io/products/pivotal-container-service to download"
```
Comment (sabbyanandan) [inline pks/init/setenv.sh]: please add: " .. and install it"

File: pks/init/setenv.sh (hunk 3 above)
Comment (chrisjs) [reply pks/init/setenv.sh]: id say we can just add that flag when we have a cluster with certs

**Comment by chrisjs** [comment]
merged in c658ea8

**Pull request closed without merging**